internal/pkg/agent/cmd/upgrade.go

```diff
@@ -53,24 +62,116 @@ func newUpgradeCommandWithArgs(_ []string, streams *cli.IOStreams) *cobra.Comman
 	cmd.Flags().String(flagPGPBytes, "", "PGP to use for package verification")
 	cmd.Flags().String(flagPGPBytesURI, "", "Path to a web location containing PGP to use for package verification")
 	cmd.Flags().String(flagPGPBytesPath, "", "Path to a file containing PGP to use for package verification")
+	cmd.Flags().BoolP(flagForce, "", false, "Advanced option to force an upgrade on a fleet managed agent")
+	err := cmd.Flags().MarkHidden(flagForce)
+	if err != nil {
+		fmt.Fprintf(streams.Err, "error while setting upgrade force flag attributes: %s", err.Error())
+		os.Exit(1)
+	}
 
 	return cmd
 }
 
+type upgradeInput struct {
+	streams   *cli.IOStreams
+	cmd       *cobra.Command
+	args      []string
+	c         client.Client
+	agentInfo client.AgentStateInfo
+	isRoot    bool
+}
+
 func upgradeCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error {
 	c := client.New()
-	return upgradeCmdWithClient(streams, cmd, args, c)
+	err := c.Connect(cmd.Context())
+	if err != nil {
+		return errors.New(err, "Failed communicating to running daemon", errors.TypeNetwork, errors.M("socket", control.Address()))
+	}
+	defer c.Disconnect()
+	state, err := c.State(cmd.Context())
+	if err != nil {
+		return fmt.Errorf("error while trying to get agent state: %w", err)
+	}
+
+	isRoot, err := utils.HasRoot()
+	if err != nil {
+		return fmt.Errorf("error while retrieving user permission: %w", err)
+	}
+
+	input := &upgradeInput{
+		streams,
+		cmd,
+		args,
+		c,
+		state.Info,
+		isRoot,
+	}
+	return upgradeCmdWithClient(input)
+}
+
+type upgradeCond struct {
+	isManaged  bool
+	force      bool
+	isRoot     bool
+	skipVerify bool
+}
+
+func checkUpgradable(cond upgradeCond) error {
+	checkManaged := func() error {
+		if cond.skipVerify {
+			return skipVeriftNotAllowedError
+		}
+
+		if !cond.isRoot {
+			return nonRootExecutionError
+		}
+
+		if !cond.force {
```

Comment:
This needs to come first so that users see that it is unsupported regardless of what they do unless they give `--force`.

This is what I see doing this locally:

```sh
❯ sudo -u elastic-agent-user elastic-development-agent upgrade 8.15.3
Error: aborting upgrade: upgrade command needs to be executed as root for fleet managed agents
For help, please see our troubleshooting guide at https://www.elastic.co/guide/en/fleet/9.0/fleet-troubleshooting.html
```

Reply:
Users should be told it is unsupported, until they supply `--force`, then they can work through the other checks.